src/context/shop-context.tsx

```diff
@@ -0,0 +1,147 @@
+import { Dispatch, createContext, useEffect, useState } from "react";
+import { ApiCart, ICart, IProductCart } from "../api/api-cart/api-cart";
+import {
+  ApiCategories,
+  ICategories,
+} from "../api/api-categories/api-categories";
+import { ApiProducts, IProduct } from "../api/api-products/api-products";
+
+export interface ContextValueInterface {
+  cart: ICart;
+}
+
+export interface IShopContext {
+  addProductToCart: (product: IProductCart) => Promise<ICart>;
+  cart?: ICart;
+  updateProductToCart: (product: IProductCart) => Promise<void>;
+  clearProductsInCart?: () => Promise<ICart>;
+  deleteProductInCart: (productUuid: string) => Promise<ICart>;
+  isActive: boolean;
+  setIsActive: Dispatch<React.SetStateAction<boolean>>;
+  categories: ICategories[];
+  quantityArray: IObj[];
+  quantityProductsCategory: (categoryUuid: string) => Promise<number>;
+  getProductsByCategory: (categoryUuid: string) => Promise<IProduct[]>;
+  endPartPath:string
+}
+export interface PropsContextModel {
+  children: JSX.Element | JSX.Element[];
+}
+
+const CartContext = createContext({} as IShopContext);
+
+interface IObj {
+  uuid: string;
+  quantity: Promise<number>;
+}
+class Obj {
```

Comment:
I hope this is a draft of PR so such classes like Obj won't be in the final version :)